src/lib/optimizer/strategy/join_ordering_rule.cpp

```diff
@@ -0,0 +1,68 @@
+#include "join_ordering_rule.hpp"
+
+#include "expression/expression_utils.hpp"
+#include "logical_query_plan/projection_node.hpp"
+#include "optimizer/dp_ccp.hpp"
+#include "optimizer/join_graph.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+JoinOrderingRule::JoinOrderingRule(const std::shared_ptr<AbstractCostEstimator>& cost_estimator)
+    : _cost_estimator(cost_estimator) {}
+
+std::string JoinOrderingRule::name() const { return "JoinOrderingRule"; }
+
+bool JoinOrderingRule::apply_to(const std::shared_ptr<AbstractLQPNode>& root) const {
+  /**
+   * Dispatch _perform_join_ordering_recursively() and fix the column order afterwards
```

Comment:
What does "fix" mean here?

Reply:
The comment below states "Join ordering might change the output column order, let's fix that". Sufficient?